src/modules/GroupChannel/context/__tests__/useMessageListScroll.spec.tsx

```diff
@@ -0,0 +1,245 @@
+import { renderHook, act, waitFor } from '@testing-library/react';
+import { useMessageListScroll } from '../hooks/useMessageListScroll';
+import { useGroupChannel } from '../hooks/useGroupChannel';
+
+jest.mock('../hooks/useGroupChannel', () => ({
+  useGroupChannel: jest.fn(),
+}));
+
+describe('useMessageListScroll', () => {
+  const mockSetIsScrollBottomReached = jest.fn();
+
+  beforeEach(() => {
+    (useGroupChannel as jest.Mock).mockImplementation(() => ({
+      state: { isScrollBottomReached: true },
+      actions: { setIsScrollBottomReached: mockSetIsScrollBottomReached },
+    }));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Initialization and Basic Behavior', () => {
+    it('should set the initial state correctly', () => {
+      const { result } = renderHook(() => useMessageListScroll('auto'));
+
+      expect(result.current.scrollRef.current).toBe(null);
+      expect(result.current.isScrollBottomReached).toBe(true);
+      expect(result.current.scrollDistanceFromBottomRef.current).toBe(0);
+      expect(result.current.scrollPositionRef.current).toBe(0);
+      expect(typeof result.current.scrollPubSub.publish).toBe('function');
+      expect(typeof result.current.scrollPubSub.subscribe).toBe('function');
+    });
+  });
+
+  describe('scrollToBottom', () => {
+    it('should call resolve only if scrollRef is null', async () => {
+      const { result } = renderHook(() => useMessageListScroll('auto'));
+      const resolveMock = jest.fn();
+
+      await act(async () => {
+        result.current.scrollPubSub.publish('scrollToBottom', { resolve: resolveMock });
+        await waitFor(() => {
+          expect(resolveMock).toHaveBeenCalled();
```

Comment:
Adding some assertion checking current scroll positions would be great here.

Reply:
Yeah added more cases here https://github.com/sendbird/sendbird-uikit-react/pull/1271/commits/d40f6b59945635e77be3b5f859852d8549ea1f87